sensorhub-core/src/main/java/org/sensorhub/impl/comm/TCPCommProvider.java

```diff
@@ -38,84 +41,79 @@
  * @author Alex Robin
  * @since July 2, 2015
  */
-public class TCPCommProvider extends AbstractModule<TCPCommProviderConfig> implements ICommProvider<TCPCommProviderConfig>
-{
+public class TCPCommProvider extends AbstractModule<TCPCommProviderConfig> implements ICommProvider<TCPCommProviderConfig> {
     static final Logger log = LoggerFactory.getLogger(TCPCommProvider.class.getSimpleName());
-    
+    private final Set<Consumer<ConnectionEventArgs>> listeners = new HashSet<>();
```

Comment:
2 Questions:

1.  Why a Set<> and why not just a simple ArrayList<>?
2. There should be a single commProvider per "driver" and a single "Consumer<ConnectionEventArgs>" pre comm provider shouldn't there? Maybe I am not thinking about this right but what is the use case for 1 CommProvider to multiple listeners?